cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/WorkflowDataset.java

```diff
@@ -0,0 +1,289 @@
+   * @throws Exception
+   */
+  public List<WorkflowRunRecord> scan(Id.Workflow id, long timeRangeStart, long timeRangeEnd) throws Exception {
+    String namespaceId = id.getApplication().getNamespaceId();
+    String applicationId = id.getApplicationId();
+    String workflowId = id.getId();
+
+    String key = namespaceId + DELIMITER + applicationId + DELIMITER + workflowId + DELIMITER;
+    byte[] startRowKey = createRowKey(key, timeRangeStart);
+    byte[] endRowKey = createRowKey(key, timeRangeEnd);
+    Scan scan = new Scan(startRowKey, endRowKey);
+
+    Scanner scanner = table.scan(scan);
+    Row indexRow;
+    List<WorkflowRunRecord> workflowRunRecordList = new ArrayList<>();
+    while ((indexRow = scanner.next()) != null) {
+      Map<byte[], byte[]> columns = indexRow.getColumns();
+      String workflowRunId = Bytes.toString(columns.get(RUNID));
+      long timeTaken = Bytes.toLong(columns.get(TIME_TAKEN));
+
+      List<ActionRuns> actionRunsList = GSON.fromJson(Bytes.toString(columns.get(NODES)), ACTION_RUNS_TYPE);
+      WorkflowRunRecord workflowRunRecord = new WorkflowRunRecord(workflowRunId, timeTaken, actionRunsList);
+      workflowRunRecordList.add(workflowRunRecord);
+    }
+    return workflowRunRecordList;
+  }
+
+  /**
+   * This method returns the statistics for a corresponding workflow.
+   * The user has to provide a time interval and a list of percentiles that are required.
+   * @param id The workflow id
+   * @param startTime The start of the time range from where the user wants the statistics
+   * @param endTime The end of the time range till where the user wants the statistics
+   * @param percentiles The list of percentiles that the user wants information on
+   * @return A statistics object that provides information about the workflow
+   * @throws Exception
+   */
+  @Nullable
+  public WorkflowStatistics getStatistics(Id.Workflow id, long startTime,
+                                                    long endTime, List<Double> percentiles) throws Exception {
+    List<WorkflowRunRecord> workflowRunRecords = scan(id, startTime, endTime);
+    int count = workflowRunRecords.size();
+
+    if (count == 0) {
+      return null;
+    }
+
+    double avgRunTime = 0.0;
+    for (WorkflowDataset.WorkflowRunRecord workflowRunRecord : workflowRunRecords) {
+      avgRunTime += workflowRunRecord.getTimeTaken();
+    }
+    avgRunTime /= count;
+
+    workflowRunRecords = sort(workflowRunRecords);
+
+    ImmutablePair<Map<String, Long>, Map<String, List<String>>> percentilesAndPercentileList =
+      getPercentiles(workflowRunRecords, percentiles);
+
+    Map<String, Long> percentileToTime = percentilesAndPercentileList.getFirst();
+    Map<String, List<String>> percentileToRunids = percentilesAndPercentileList.getSecond();
+
+    Map<String, List<Long>> actionToRunRecord = getActionRuns(workflowRunRecords);
+
+    Map<String, Map<String, Double>> actionToStatistic = new HashMap<>();
```

Comment:
Isn't it the case that the `actionToStatistic` map will only hold integer numbers (non fraction) as the value?
In that case, it makes sense just to use long instead of double as the value, right?